### DISCUSSION TITLE
Export format of Fossify's vCards: 3.0?  4.0?  Use of TYPE= parameter?

### DISCUSSION THREAD

**Discussion by DiagonalArg** [FossifyOrg/General-Discussion#660] [General]
I'm trying to write a script to translate Google vCards to something Fossify understands and, more broadly, to understand the vCard spec(s).  For that I need to understand what Fossify is doing behind the scenes:

1. I have a file of some 2000 vCards exported from Google.  They are labeled VERSION:3.0, though they have google specific extensions.  I import those cards to Fossify and then immediately export.  I see that the result is labeled VERSION:4.0 (and the Google extensions are missed, but that's just a side note).  My first question is why, starting from no contacts in Fossify, when I create a single contact and export it, it exports as VERSION:3.0 instead?  What is Fossify doing here?

```
BEGIN:VCARD
VERSION:3.0
PRODID:ez-vcard 0.11.3
FN:Pre First Middle Last Suffix
N:Last;First;Middle;Pre;Suffix
NICKNAME:Nickname
TEL;TYPE=Landline:555-1215
TEL;TYPE=HOME:+1.510-555-6666
EMAIL;TYPE=HOME:foo@bar.com
EMAIL;TYPE=Jason:foo@bar.com
BDAY:2024-01-01
ADR;TYPE=Custom-Label:;;300 Amsterdam Av.\, New York\, NY 10027-2819
ADR;TYPE=HOME:;;300 Amsterdam Av.\, New York\, NY 10027-2819
IMPP:skype:skypename
NOTE:Notes located\nHere!
ORG:Company
TITLE:Title
URL:http://www.foobar.com
CATEGORIES:AaaaaLabel
END:VCARD
```

2. Going to [vCard Elements in IANA](https://www.iana.org/assignments/vcard-elements/vcard-elements.xhtml), I see that TYPE= parameter can have only a limited set of values.  eg. for the EMAIL property, it's only "work" and "home".  Fossify appears to be using TYPE="any string", as you see above.  Is there some spec conformant way of doing this properly?  Besides IANA, these appear to be the relevant RFC's.  Please let me know if I've missed anything:

4.0: https://datatracker.ietf.org/doc/html/rfc6350
4.0, Format Extension: https://www.rfc-editor.org/rfc/rfc6715.html
4.0 extension for character escaping; https://datatracker.ietf.org/doc/html/rfc6868
4.0 extension for JSContact: https://datatracker.ietf.org/doc/html/rfc9554

As a comparison, this is how the Stock AOSP contacts app present it, using "X-Landline" for example.

```
BEGIN:VCARD
VERSION:2.1
N:Last;First;Middle;Pre;Suffix
FN:Pre First Middle Last, Suffix
X-PHONETIC-FIRST-NAME:PFirst
X-PHONETIC-MIDDLE-NAME:PMiddle
X-PHONETIC-LAST-NAME:PLast
TEL;HOME:+1-510-555-6666
TEL;X-Landline:555-1215
X-ANDROID-CUSTOM:vnd.android.cursor.item/nickname;Nickname;1;;;;;;;;;;;;;
EMAIL;HOME:foo@bar.com
ADR;X-Home:;;300 Amsterdam Av., New York, NY 10027-2819;;;;
ADR;X-Custom-Label:;;300 Amsterdam Av., New York, NY 10027-2819;;;;
ORG:Company\;Department
TITLE:Title
URL:http://www.foobar.com
NOTE;ENCODING=QUOTED-PRINTABLE:=4E=6F=74=65=73=20=6C=6F=63=61=74=65=64=0A=48=65=72=65=21
END:VCARD
```

**Comment by naveensingh** [comment]
The share contact option exports 3.0. Automatic backups and exports from settings do 4.0.

The inconsistent behavior is intentional as per https://github.com/SimpleMobileTools/Simple-Contacts/pull/995, but I'm thinking of resetting it back to 4.0.

**Comment by DiagonalArg** [comment]
@naveensingh / could you respond regarding the specs in relation to the TYPE= parameter?

**Reply by naveensingh** [reply]
You're correct there. The app falls back to a non-standard label (without `X-` prefix) when it can't find a proper type. 

It'll be fixed, but feel free to file a bug report.
